src/Nethermind/Nethermind.Optimism/CL/L1Bridge/EthereumL1Bridge.cs

```diff
@@ -269,29 +250,21 @@ public async Task<ReceiptForRpc[]> GetReceiptsByBlockHash(Hash256 blockHash, Can
         return result;
     }
 
-    private async Task<L1Block> GetHead(CancellationToken token)
-    {
-        L1Block? result = await _ethL1Api.GetHead(true);
-        while (result is null)
-        {
-            token.ThrowIfCancellationRequested();
-            _logger.Warn($"Unable to get L1 head");
-            result = await _ethL1Api.GetHead(true);
-        }
+    private async Task<L1Block> GetHead(CancellationToken token) =>
+        await RetryGetBlock(async () => await _ethL1Api.GetHead(true), token);
 
-        return result.Value;
-    }
+    private async Task<L1Block> GetFinalized(CancellationToken token) =>
+        await RetryGetBlock(async () => await _ethL1Api.GetFinalized(true), token);
 
-    private async Task<L1Block> GetFinalized(CancellationToken token)
+    private async Task<L1Block> RetryGetBlock(Func<Task<L1Block?>> getBlock, CancellationToken token)
     {
-        L1Block? result = await _ethL1Api.GetFinalized(true);
+        L1Block? result = await getBlock();
         while (result is null)
```

Comment:
`while !token.CancellationRequested`?

Reply:
I want to throw for type consistency purposes, so I don't need to return nullable type.